### PR TITLE
Add solver

### PR DESCRIPTION
**Pull request by Cameron-Taleghani**
Added a `solver` method that adds the text `"Click me!"` to a button so that when all of said buttons are clicked the puzzle is solved. This method is invoked in the constructor and whenever a button is clicked. Also made some whitespace and style changes: removed some whitespace lines, added whitespace to `if(...)` and `for(...)`so that it is `if (...)` and `for (...)`, and moved the invocations of `push` onto the same line where appropriate.

### CODE REVIEW

File: lights-out.js [hunk 1]
```diff
@@ -40,28 +39,60 @@ class LightsOut {
...
                         const [r, c] = neighbor;
                         document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]`)
-                            .classList.toggle("on")
+                            .classList.toggle("on");
+                    }
+                    if (!button.textContent) {
+                        this.solve();
+                    } else {
+                        button.textContent = null;
                     }
```
Comment (mikegagnon) [inline lights-out.js]: I'm pretty sure you can remove this conditional (if and else) and simply replace it with `this.solve();`.

File: lights-out.js [hunk 2]
```diff
@@ -40,28 +39,60 @@ class LightsOut {
...
+        for (let row = 0; row < this.numRows; row++) {
+            for (let col = 0; col < this.numCols; col++) {
+                const button = document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`);
+                const neighbors = this.getNeighbors(row, col);
+                if (button.classList.contains("on")) { 
+                    if (button.textContent) {
+                        button.textContent = null;
+                    } else {
+                        button.textContent = "Click me!";
+                    }
```
Comment (mikegagnon) [inline lights-out.js]: I think you can remove this conditional by adding a new line after `const neighbors = ...`. The new line should be `neighbors.push([row, col]);`

This insight makes me wonder if we would have been better off to make `getNeighbors(row, col) {...}` include `[row, col]` in its return result. However, we are close to to finishing this project and we are past the good-idea cutoff point, so let's leave `getNeighbors(row, col) {...}` as is.

**Review by mikegagnon — CHANGES_REQUESTED** [review]
Good work. Just a few small changes requested.

File: lights-out.js (hunk 1 above)
Comment (Cameron-Taleghani) [reply lights-out.js]: Done

File: lights-out.js (hunk 2 above)
Comment (Cameron-Taleghani) [reply lights-out.js]: This doesn't seem to be working, unless I am misunderstanding. I am deleting the conditional

```
if (button.classList.contains("on")) { 
    if (button.textContent) {
        button.textContent = null;
    } else {
        button.textContent = "Click me!";
    }
``` 
and adding `neighbors.push([row, col]);` after `const neighbors = this.getNeighbors(row, col);`. 

Side note, there are actually two lines of `const neighbors = this.getNeighbors(row, col);`, so one is redundant.

Comment (mikegagnon) [reply lights-out.js]: Let's discuss this issue during our meeting at 11am. For future reference, code reviews aren't the best place to collaboratively debug code. It's better to chat on other platforms, such as Discord. This way, it's easier for our teammates from the future to study the history of the code of this project.

Comment (Cameron-Taleghani) [reply lights-out.js]: Gotcha

**Review by mikegagnon — APPROVED** [review]
Good work.

**Pull request merged**